phys2bids/phys2bids.py

```diff
@@ -119,160 +103,111 @@ def _main(argv=None):
-
     utils.check_file_exists(infile)
-    print('File exists')
+    outfile = os.path.join(options.outdir, os.path.basename(options.filename[:-4]))
 
-    # Read infos from file
+    # Read file!
     if ftype == 'acq':
-        from bioread import read_file
-
-        data = read_file(infile).channels
-        print_info_acq(options.filename, data)
+        from phys2bids.interfaces.acq import populate_phys_input
     elif ftype == 'txt':
-        header = print_info_txt(options.filename)
+        raise Exception('txt not yet supported')
+    else:
+        raise Exception('This shouldn\'t happen, check out the last few '
+                        'lines of code')
+
+    print('Reading the file')
+    phys_in = populate_phys_input(infile, options.chtrig)
+    print('Reading infos')
+    utils.print_info(options.filename, phys_in)
+    # #!# Here the function viz.plot_channel should be called
+    # for the desired channels.
 
     # If file has to be processed, process it
     if not options.info:
+
+        # #!# Get option of no trigger! (which is wrong practice or Respiract)
+        phys_in.check_trigger_amount(options.thr, options.num_tps_expected,
+                                     options.tr)
+        print('Checking that the output folder exists')
+        utils.path_exists_or_make_it(options.outdir)
+        print('Plot trigger')
+        viz.plot_trigger(phys_in.timeseries[0], phys_in.timeseries[1],
+                         outfile, options)
+
+        # The next few lines remove the undesired channels from phys_in.
+        if options.chsel:
+            print('Dropping unselected channels')
+            for i in [x for x in reversed(range(0, phys_in.ch_amount))
+                      if x not in options.chsel]:
+                phys_in.delete_at_index(i)
+
+        # If requested, change channel names.
+        if options.ch_name:
+            print('Renaming channels with given names')
+            phys_in.rename_channels(options.ch_name)
+
+        # The next few lines create a dictionary of different blueprint_input
+        # objects, one for each unique frequency in phys_in
+        uniq_freq_list = set(phys_in.freq)
+        print(f'Found {len(uniq_freq_list)} unique frequencies.')
+        phys_out = {}
+        for uniq_freq in uniq_freq_list:
+            phys_out[uniq_freq] = deepcopy(phys_in)
+            for i in [i for i, x in enumerate(reversed(phys_in.freq))
+                      if x != uniq_freq]:
+                phys_out[uniq_freq].delete_at_index(phys_in.ch_amount-i-1)
+
+        for uniq_freq in uniq_freq_list:
+            phys_out[uniq_freq] = blueprint_output.init_from_blueprint(phys_out[uniq_freq])
+
+        output_amount = len(uniq_freq_list)
+        if output_amount > 1:
+            print(f'Found {output_amount} different frequencies in input!\n'
+                  f'Consequently, preparing {output_amount} of output files')
```

Comment:
This is a matter of tastes but imo the word _Consequently_ should not be there. It just sounds weird to me from an end-user pov.